### DISCUSSION TITLE
Gosund SP112 keeps resetting itself

### DISCUSSION THREAD

**Discussion by AleXSR700** [arendst/Tasmota#12057] [Q&A] [answered]
Hello everyone,
I just flashed Tasmota onto my first of two Gosund SP112 devices and for some reason it keeps resetting itself to a state where it enables the Tasmota Wifi and I have to enter the wifi details and of course afterwards the template information again.

Does anybody know what could be causing this?

The power button also has no functionality.

Maybe the wrong template, eventough I tried the 112 and 112v3.4 template?

Thank you for your feedback :-)

Alex

**Comment by AleXSR700** [comment, marked as answer]
Good morning everyone,

I of course kept trying and working on fixing the issue and have gotten one step further. But now I am really stuck.

I was only able to stop the resetting/rebooting with SetOption1 1. SetOption65 1 and SerialLog 0 did nothing.

I have tried the configurations

{"NAME":"Gosund 112v3.4","GPIO":[56,0,57,0,132,134,0,0,131,30,21,0,0],"FLAG":4,"BASE":18}

and

{"NAME":"SP112","GPIO":[321,0,320,0,2656,2720,0,0,2624,257,224,0,0,4800],"FLAG":0,"BASE":18}

But neither allows me to turn USB on/off. Only main outlet. The 112v3.4 turns on the blue light when I switch on main and the 112 turns on the red light when I turn on main.
Power button does absolutely nothing. Also shows no response in console when I push it (with Weblog set to 4).

Is it possible to tell which of the two should be correct and how to enable USB switching? USB is permantently turned on right now.

On the inside it reads
Model SP112-1 and on the mainboard itself:
SP112_Main_V2.8_10A&16A
20190315
So presumably the old model and not the 3.4?

The energy readings also do not work. They are all stuck at 0. All of them. I did not calibrate because I first wanted to check that everything was working, but at least Voltage should not be 0.

**Reply by Jason2866** [reply]
Did you disabled seriallog as recommended? https://templates.blakadder.com/gosund_SP112.html
Try this template https://templates.blakadder.com/blitzwolf_SHP5.html

**Reply by AleXSR700** [reply]
As mentioned, yes, I did.

In the end I was able to figure it out. The SHP5 template works for the Gosund SP112 whereas the SP112 and SP112v3.4 do not.
So if anyone experiences the same issue, just ignore the name and use the Blitzwolf SHP5 template :-)
